### PR TITLE
feat(DTFS2-6182): replace get facilities graphql endpoint with REST endpoint

### PR DESCRIPTION
**Pull request by francescastocco**
## Introduction
We need to standardise the TFM API by moving away from having both gql and REST endpoints. We will move the gql endpoints to REST.

## Resolution
Replace the get facilities graphql endpoint with REST endpoint as part of the gql -> rest migration in TFM API.

### CODE REVIEW

File: trade-finance-manager-api/src/v1/controllers/facility.controller.js [hunk 1]
```diff
@@ -22,6 +27,66 @@ const getFacility = async (req, res) => {
...
+
+      if (!facility.ukefFacilityId) {
+        facility.ukefFacilityId = '-';
+      }
+
+      const latestCompletedAmendment = findLatestCompletedAmendment(dbFacility.amendments);
+      let facilityCoverEndDate = '';
+      let facilityCoverEndDateEpoch = '';
+
+      if (REGEX.DATE.test(facility.coverEndDate) && facility.coverEndDate) {
```
Comment (ChristianMcc) [inline trade-finance-manager-api/src/v1/controllers/facility.controller.js:51]: Is this regex just the way it was done before? It also feels wrong to check facility.coverEndDate exists after testing it in regex before that
do you reckon we can do something instead where we create a new date from the coverEndDate and check that the date is valid after instead? (sorry I'm reviewing copy pasted code but worth having these noted)

File: trade-finance-manager-ui/server/controllers/facilities/index.js [hunk 2]
```diff
@@ -3,7 +3,8 @@ const CONSTANTS = require('../../constants');
 
 const getFacilities = async (req, res) => {
   const { userToken } = req.session;
-  const apiResponse = await api.getFacilities();
+  // queryParams here?
```
Comment (ChristianMcc) [inline trade-finance-manager-ui/server/controllers/facilities/index.js:6]: Interesting that they never passed some parameters through here before - I'd leave a comment saying query parameters have been implemented in APIs but aren't used in the frontends currently

File: trade-finance-manager-api/src/v1/controllers/facility.controller.js (hunk 1 above)
Comment (francescastocco) [reply trade-finance-manager-api/src/v1/controllers/facility.controller.js]: Ah yes, I did refactor this a bit, I've now updated the condition which doesn't make sense - [this](https://github.com/UK-Export-Finance/dtfs2/blob/main/trade-finance-manager-api/src/graphql/resolvers/query-facilities.js) is the code that I am replicating

File: trade-finance-manager-api/src/v1/controllers/facility.controller.js [hunk 3]
```diff
@@ -1,4 +1,105 @@
...
+  try {
+    const queryParams = req.body;
+
+    if (!queryParams) {
+      return {};
+    }
+
+    const dbFacilities = await api.getAllFacilities(queryParams);
+
+    const facilities = dbFacilities.map((dbFacility) => {
```
Comment (AlexBramhill) [inline trade-finance-manager-api/src/v1/controllers/facility.controller.js:40]: Could this be a separate function?

File: trade-finance-manager-ui/server/controllers/facilities/index.js [hunk 4]
```diff
@@ -3,7 +3,8 @@ const CONSTANTS = require('../../constants');
 
 const getFacilities = async (req, res) => {
   const { userToken } = req.session;
-  const apiResponse = await api.getFacilities();
+  // Query parameters have ben implemented in APIs but aren't currently used in frontends
```
Comment (AlexBramhill) [inline trade-finance-manager-ui/server/controllers/facilities/index.js:6]:
```suggestion
  // Query parameters have been implemented in APIs but aren't currently used in frontends
```

**Review by AlexBramhill — COMMENTED** [review]
reviewed non-mapping files

File: trade-finance-manager-api/src/v1/controllers/facility.controller.js [hunk 5]
```diff
@@ -1,4 +1,105 @@
...
+    return res.status(500).send(error.message);
+  }
+};
+
+const getFacilities = async (req, res) => {
+  try {
+    const queryParams = req.body;
+
+    if (!queryParams) {
+      return {};
```
[the remark this answers is not in the document]
Comment (AlexBramhill) [reply trade-finance-manager-api/src/v1/controllers/facility.controller.js:35]: I've added a todo to check this -- It's in the existing resolver implimentation

**Pull request merged**